plugins/flytekit-greatexpectations/flytekitplugins/great_expectations/task.py

```diff
@@ -0,0 +1,275 @@
+import datetime
+import logging
+import os
+import shutil
+from dataclasses import dataclass
+from typing import Any, Dict, List, Optional, Type, Union
+
+import great_expectations as ge
+from dataclasses_json import dataclass_json
+from great_expectations.checkpoint import SimpleCheckpoint
+from great_expectations.core.run_identifier import RunIdentifier
+from great_expectations.core.util import convert_to_json_serializable
+from great_expectations.exceptions import ValidationError
+
+from flytekit import PythonInstanceTask
+from flytekit.core.context_manager import FlyteContext
+from flytekit.extend import Interface
+from flytekit.types.file.file import FlyteFile
+from flytekit.types.schema import FlyteSchema
+
+
+@dataclass_json
+@dataclass
+class BatchRequestConfig(object):
+    """
+    Use this configuration to configure Batch Request. A BatchRequest can either be
+    a simple BatchRequest or a RuntimeBatchRequest.
+
+    Args:
+        data_connector_query: query to request data batch
+        runtime_parameters: parameters to be passed at runtime
+        batch_identifiers: identifiers to identify the data batch
+        batch_spec_passthrough: reader method if your file doesn't have an extension
+    """
+
+    data_connector_query: Optional[Dict[str, Any]] = None
+    runtime_parameters: Optional[Dict[str, Any]] = None
+    batch_identifiers: Optional[Dict[str, str]] = None
+    batch_spec_passthrough: Optional[Dict[str, Any]] = None
+
+
+class GreatExpectationsTask(PythonInstanceTask[BatchRequestConfig]):
+    """
+    This task can be used to validate your data.
```

Comment:
lets write a little more doc here to explain - usecase or when can you use it

Reply:
I have added two more lines explaining the use case.